fix: balance calculations to be compatible with new chain changes

0oM4R opened the pull request:
### Description
To get the total balance now we have to add the free and the reserved, locked balance is not used anymore

### Changes

- replaced locked with reserved
- show the sum of reserved and free in profile manager
- make `freeBalance` variable computed to make it always synced with the balanc

### Related Issues

- #3340 

### Tested Scenarios
- having zero tft, and check the profile manager

- use polkadot UI and call `system/account` to retrieve the actual balance and compare it with the shown in profile manager
- try to empty you account from balance at all and check for the shown values 
- also try to deploy when you have balance less than 2 TFT < should fail> 
- try to deploy when the **free** balance is larger than 2 TFT 

![Screenshot from 2024-09-10 11-47-17](https://github.com/user-attachments/assets/57f76400-5ffc-4e8d-abef-2865a8005732)

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

0oM4R committed:
fix: balance caculations
adjust caculating balance to be compatible with new tfchain changes

0oM4R committed:
fix: build

amiraabouhadid reviewed (commented):
works when balance is exactly 2tft
![image](https://github.com/user-attachments/assets/bf5e12a3-05fc-47f4-88ec-e64ba72638f7)

0oM4R commented:
> works when balance is exactly 2tft ![image](https://private-user-images.githubusercontent.com/56790126/368500617-19a9ee97-acd5-462b-8cc1-bb6a321aa16a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY2NTEyODgsIm5iZiI6MTcyNjY1MDk4OCwicGF0aCI6Ii81Njc5MDEyNi8zNjg1MDA2MTctMTlhOWVlOTctYWNkNS00NjJiLThjYzEtYmI2YTMyMWFhMTZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE4VDA5MTYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5Nzc5NDg3YmNhYzQyNDA0N2NjOWE2MDBmMWU4Mzc4NjU4ZWJhODhkMGY4NTY0YzFhNTExYWE4NzZhNzMzYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Pi4E4W1lnjT0ivoOtA4MeWht0JaqpCBsTqkD8EvOKwk)

this is the expected behavior right ?

amiraabouhadid reviewed (changes requested):
however, rounding the numbers (1.998 is rounded to 2) is confusing in the err msg . I suggest you use the same number of decimals in err msg as used in balance display
![image](https://github.com/user-attachments/assets/0821050e-50c8-499a-a83a-9417ced8bc0a)

0oM4R committed:
Fix: weblet layout error msg
remove the current balance as the showing the numbers with floor, tofixed, trunc will be misleading

0oM4R committed:
Fix: show zero if the balance is zero in profile manager

0oM4R commented:
![image](https://github.com/user-attachments/assets/a3a96b7a-9843-404d-b17e-f8bd863ae039)

amiraabouhadid reviewed (approved):
tested with less than 2TFT, works
![Screenshot from 2024-09-22 12-23-55](https://github.com/user-attachments/assets/a132ebb7-a780-4255-83ac-6607302b6dfd)tested with exactly 2TFT, works
![image](https://github.com/user-attachments/assets/691ed3c5-d4be-46dc-b1ff-48f85de94dde)
